martian/src/lib.rs

```diff
@@ -286,48 +284,38 @@ fn report_error(md: &mut Metadata, e: &Error, is_assert: bool) {
     let _ = write_errors(&format!("{e:#}"), is_assert);
 }
 
-fn get_generator_name() -> String {
-    std::env::var("CARGO_BIN_NAME")
-        .or_else(|_| std::env::var("CARGO_CRATE_NAME"))
-        .or_else(|_| std::env::var("CARGO_PKG_NAME"))
-        .unwrap_or_else(|_| {
-            option_env!("CARGO_BIN_NAME")
-                .or(option_env!("CARGO_CRATE_NAME"))
-                .unwrap_or("martian-rust")
-                .into()
-        })
-}
-
+/// Write MRO to filename or stdout.
 pub fn martian_make_mro(
     header_comment: &str,
-    file_name: Option<impl AsRef<Path>>,
+    filename: Option<impl AsRef<Path>>,
     rewrite: bool,
     mro_registry: Vec<StageMro>,
-) -> Result<(), Error> {
-    if let Some(ref f) = file_name {
-        let file_path = f.as_ref();
-        if file_path.is_dir() {
-            return Err(format_err!(
-                "Error! Path {} is a directory!",
-                file_path.display()
-            ));
-        }
-        if file_path.exists() && !rewrite {
-            return Err(format_err!(
-                "File {} exists. You need to explicitly mention if it is okay to rewrite.",
-                file_path.display()
-            ));
-        }
+) -> Result<()> {
+    if let Some(filename) = &filename {
+        let filename = filename.as_ref();
+        ensure!(
+            !filename.is_dir(),
+            "Path {} is a directory",
+            filename.display()
+        );
+        ensure!(
+            rewrite || !filename.exists(),
+            "File {} exists. Use --rewrite to overwrite it.",
```

Comment:
`--rewrite` is a caller specific detail, but this is valid for all our use cases